### PR TITLE
doc refactor to master

### PR DESCRIPTION
**Pull request by leckie-chn**
* activate homepage on nni.readthedocs.io
* disable homepage on github pages
* will include new docs after merge

### CODE REVIEW

File: examples/trials/mnist/mnist_before.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    try:
+        params = {
+            'data_dir': '/tmp/tensorflow/mnist/input_data',
+            'dropout_rate': 0.5,
+            'channel_1_num': 32,
+            'channel_2_num': 64,
+            'conv_size': 5,
+            'pool_size': 2,
+            'hidden_size': 1024,
+            'learning_rate': 1e-4,
```
Comment (chicm-ms) [inline examples/trials/mnist/mnist_before.py]: There are some updates on mnist example recently,  could you sync from master mnist example again?

Comment (leckie-chn) [reply examples/trials/mnist/mnist_before.py]: see https://github.com/Microsoft/nni/pull/648/commits/d6bbb79c92f706711b7d42c65b0af790a8415b8e

File: docs/Builtin_Tuner.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+
+<a name="Evolution"></a>
+
+![](https://placehold.it/15/1589F0/000000?text=+) `Naive Evolution`
+
+> Builtin Tuner Name: **Evolution**
+
+**Suggested scenario**
+
+Its requirement of computation resource is relatively high. Specifically, it requires large inital population to avoid falling into local optimum. If your trial is short or leverages assessor, this tuner is a good choice. And, it is more suggested when your trial code supports weight transfer, that is, the trial could inherit the converged weights from its parent(s). This can greatly speed up the training progress.
```
Comment (chicm-ms) [inline docs/Builtin_Tuner.md]: typo:  inital   => initial ?

File: docs/Builtin_Tuner.md [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+**Suggested scenario**
+
+It is suggested when you have limited computation resource but have relatively large search space. It performs good in the scenario that intermediate result (e.g., accuracy) can reflect good or bad of final result (e.g., accuracy) to some extent.
+
+**Requirement of classArg**
+
+* **optimize_mode** (*maximize or minimize, optional, default = maximize*) - If 'maximize', tuners will return the hyperparameter set with larger expectation. If 'minimize', tuner will return the hyperparameter set with smaller expectation.
+* **R** (*int, optional, default = 60*) - the maximum STEPS (could be the number of mini-batches or epochs) can be allocated to a trial. Each trial should use STEPS to control how long it runs.
+* **eta** (*int, optional, default = 3*) - proportion of discarded trials
```
Comment (chicm-ms) [inline docs/Builtin_Tuner.md]: I do not understand,  default is 3,  how 3 can be a proportion of discarded trials?

File: docs/AnnotationSpec.md [hunk 4]
```diff
@@ -1,58 +1,70 @@
...
 
- 2. Annotation variable in code as:
+Below is an example:
 
-    '''@nni.variable(nni.choice(2,3,5,7),name=self.conv_size)'''
+```python
+'''@nni.variable(nni.choice(0.1, 0.01, 0.001), name=learning_rate)'''
+learning_rate = 0.1
+```
+The meaning of this example is that NNI will choose one of several values (0.1, 0.01, 0.001) to assign to the learning_rate variable. Specifically, this first line is an NNI annotation, which is a single string. Following is an assignment statement. What nni does here is to replace the right value of this assignment statement according to the information provided by the annotation line.
```
Comment (xuehui1991) [inline docs/AnnotationSpec.md]: "a NNI annotation"

File: docs/AnnotationSpec.md [hunk 5]
```diff
@@ -1,58 +1,70 @@
...
+`@nni.variable` is used in NNI to annotate a variable.
 
- 1. `@nni.variable(nni.choice(option1,option2,...,optionN),name=variable)`  
-    Which means the variable value is one of the options, which should be a list The elements of options can themselves be stochastic expressions
+**Arguments**
 
- 2. `@nni.variable(nni.randint(upper),name=variable)`  
-    Which means the variable value is a random integer in the range [0, upper).
+- **sampling_algo**: Sampling algorithm that specifies a search space. User should replace it with a built-in NNI sampling function whose name consists of an `nni.` identification and a search space type specified in [SearchSpaceSpec](SearchSpaceSpec.md) such as `choice` or `uninform`. 
+- **name**: The name of the variable that the selected value will be assigned to. Note that this argument should be the same as the left value of the following assignment statement.
```
Comment (xuehui1991) [inline docs/AnnotationSpec.md]: "uniform"

File: docs/AnnotationSpec.md [hunk 6]
```diff
@@ -1,58 +1,70 @@
...
-    Which means the variable value is a real value that's normally-distributed with mean mu and standard deviation sigma.
+`@nni.function_choice` is used to choose one from several functions.
 
- 8. `@nni.variable(nni.qnormal(label, mu, sigma, q),name=variable)`  
-    Which means the variable value is a value like round(normal(mu, sigma) / q) * q
+**Arguments**
 
- 9. `@nni.variable(nni.lognormal(label, mu, sigma),name=variable)`  
-    Which means the variable value is a value drawn according to exp(normal(mu, sigma))
+- **\*functions**: Several functions that are waiting to be selected from. Note that it should be a complete function call with aruguments. Such as `max_pool(hidden_layer, pool_size)`.
```
Comment (xuehui1991) [inline docs/AnnotationSpec.md]: "aruguments"? -- > "arguments"

File: docs/AnnotationSpec.md [hunk 7]
```diff
@@ -1,58 +1,70 @@
...
+- **name**: The name of the function that will be replace in the following assignment statement.
```
Comment (xuehui1991) [inline docs/AnnotationSpec.md]: "be replace"  --> "be replaced"?

File: docs/AnnotationSpec.md [hunk 8]
```diff
@@ -1,58 +1,70 @@
...
+```python
+"""@nni.function_choice(max_pool(hidden_layer, pool_size), avg_pool(hidden_layer, pool_size), name=max_pool)"""
+h_pooling = max_pool(hidden_layer, pool_size)
+```
+
+### 3. Annotate intermediate result
+
+   `'''@nni.report_intermediate_result(metrics)'''`
+
+`@nni.report_intermediate_result` is used to report itermediate result, whose usage is the same as `nni.report_intermediate_result` in [Trials.md](Trials.md)
```
Comment (xuehui1991) [inline docs/AnnotationSpec.md]: "itermediate " --> "intermediate "

File: docs/AnnotationSpec.md [hunk 9]
```diff
@@ -1,58 +1,70 @@
...
+
+### 4. Annotate final result
+
+   `'''@nni.report_final_result(metrics)'''`
+
+`@nni.report_final_result` is used to report final result of the current trial, whose usage is the same as `nni.report_final_result` in [Trials.md](Trials.md)
```
Comment (xuehui1991) [inline docs/AnnotationSpec.md]: "report final result " --> "report the final result "??

File: docs/Builtin_Assessors.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
+# Builtin Assessors
```
Comment (xuehui1991) [inline docs/Builtin_Assessors.md]: "Builtin" ---> "Built-in" ??

File: docs/Builtin_Assessors.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+|[Curvefitting](https://github.com/Microsoft/nni/blob/master/src/sdk/pynni/nni/curvefitting_assessor/README.md)<br>[(Usage)](#Curvefitting)|Curve Fitting Assessor is a LPA(learning, predicting, assessing) algorithm. It stops a pending trial X at step S if the prediction of final epoch's performance worse than the best final performance in the trial history. In this algorithm, we use 12 curves to fit the accuracy curve|It is applicable in a wide range of performance curves, thus, can be used in various scenarios to speed up the tuning progress. Even better, it's able to handle and assess curves with similar performance.|
+
+<br>
+
+## Usage of Builtin Assessors
+
+Use builtin assessors provided by NNI sdk requires to declare the  **builtinAssessorName** and **classArgs** in `config.yml` file. In this part, we will introduce the detailed usage about the suggested scenarios, classArg requirements, and example for each assessor.
```
Comment (xuehui1991) [inline docs/Builtin_Assessors.md]: "NNI sdk requires"  --> "NNI SDK requires" ??

File: docs/Builtin_Tuner.md [hunk 12]
```diff
@@ -0,0 +1,313 @@
+# Builtin Tuners
```
Comment (xuehui1991) [inline docs/Builtin_Tuner.md]: "Builtin Tuner" --> "Built-in Tuner"???

File: docs/Builtin_Tuner.md [hunk 13]
```diff
@@ -0,0 +1,313 @@
...
+|**Anneal**<br>[(Usage)](#Anneal)|This simple annealing algorithm begins by sampling from the prior, but tends over time to sample from points closer and closer to the best ones observed. This algorithm is a simple variation on the random search that leverages smoothness in the response surface. The annealing rate is not adaptive.|
+|**Naive Evolution**<br>[(Usage)](#Evolution)|Naive Evolution comes from Large-Scale Evolution of Image Classifiers. It randomly initializes a population-based on search space. For each generation, it chooses better ones and does some mutation (e.g., change a hyperparameter, add/remove one layer) on them to get the next generation. Naive Evolution requires many trials to works, but it's very simple and easy to expand new features.|
+|**SMAC**<br>[(Usage)](#SMAC)|SMAC is based on Sequential Model-Based Optimization (SMBO). It adapts the most prominent previously used model class (Gaussian stochastic process models) and introduces the model class of random forests to SMBO, in order to handle categorical parameters. The SMAC supported by nni is a wrapper on the SMAC3 github repo.|
```
Comment (xuehui1991) [inline docs/Builtin_Tuner.md]: "SMAC3 github repo"  --> "SMAC3 Github repo"

File: docs/Builtin_Tuner.md [hunk 14]
```diff
@@ -0,0 +1,313 @@
...
+|[Network Morphism](https://github.com/Microsoft/nni/blob/master/src/sdk/pynni/nni/networkmorphism_tuner/README.md)<br>[(Usage)](#NetworkMorphism)|Network Morphism provides functions to automatically search for architecture of deep learning models. Every child network inherits the knowledge from its parent network and morphs into diverse types of networks, including changes of depth, width, and skip-connection. Next, it estimates the value of a child network using the historic architecture and metric pairs. Then it selects the most promising one to train.|
+|**Metis Tuner**<br>[(Usage)](#MetisTuner)|Metis offers the following benefits when it comes to tuning parameters: While most tools only predicts the optimal configuration, Metis gives you two outputs: (a) current prediction of optimal configuration, and (b) suggestion for the next trial. No more guesswork. While most tools assume training datasets do not have noisy data, Metis actually tells you if you need to re-sample a particular hyper-parameter.|
```
Comment (xuehui1991) [inline docs/Builtin_Tuner.md]: "only predicts " --->"only predict "

File: docs/Builtin_Tuner.md [hunk 15]
```diff
@@ -0,0 +1,313 @@
...
+
+<br>
+
+## Usage of Builtin Tuners
+
+Use builtin tuner provided by NNI sdk requires to declare the  **builtinTunerName** and **classArgs** in `config.yml` file. In this part, we will introduce the detailed usage about the suggested scenarios, classArg requirments and example for each tuner.
```
Comment (xuehui1991) [inline docs/Builtin_Tuner.md]: "NNI sdk requires" --->"NNI SDK requires"?

File: docs/Builtin_Tuner.md [hunk 16]
```diff
@@ -0,0 +1,313 @@
...
+|[Network Morphism](https://github.com/Microsoft/nni/blob/master/src/sdk/pynni/nni/networkmorphism_tuner/README.md)<br>[(Usage)](#NetworkMorphism)|Network Morphism provides functions to automatically search for architecture of deep learning models. Every child network inherits the knowledge from its parent network and morphs into diverse types of networks, including changes of depth, width, and skip-connection. Next, it estimates the value of a child network using the historic architecture and metric pairs. Then it selects the most promising one to train.|
+|**Metis Tuner**<br>[(Usage)](#MetisTuner)|Metis offers the following benefits when it comes to tuning parameters: While most tools only predicts the optimal configuration, Metis gives you two outputs: (a) current prediction of optimal configuration, and (b) suggestion for the next trial. No more guesswork. While most tools assume training datasets do not have noisy data, Metis actually tells you if you need to re-sample a particular hyper-parameter.|
+
+<br>
+
+## Usage of Builtin Tuners
+
```
Comment (xuehui1991) [inline docs/Builtin_Tuner.md]: "requirments"  --> "requirements"???

File: docs/Builtin_Tuner.md (hunk 2 above)
Comment (xuehui1991) [inline docs/Builtin_Tuner.md]: > "inital

 "  --> "inital"

File: docs/Builtin_Tuner.md [hunk 17]
```diff
@@ -0,0 +1,313 @@
...
+<a name="Hyperband"></a>
+
+![](https://placehold.it/15/1589F0/000000?text=+) `Hyperband`
+
+> Builtin Advisor Name: **Hyperband**
+
+**Suggested scenario**
+
+It is suggested when you have limited computation resource but have relatively large search space. It performs good in the scenario that intermediate result (e.g., accuracy) can reflect good or bad of final result (e.g., accuracy) to some extent.
```
Comment (xuehui1991) [inline docs/Builtin_Tuner.md]: “It performs good” --- 》“It performs well”

File: docs/Builtin_Assessors.md [hunk 18]
```diff
@@ -0,0 +1,74 @@
+# Builtin Assessors
+
+NNI provides the-state-of-art tuning algorithm in our builtin-assessors and makes them easy to use. Below is the brief overview of NNI current builtin Assessors:
```
Comment (chicm-ms) [inline docs/Builtin_Assessors.md]: the-state-of-art   => state-of-the-art ?

File: docs/Customize_Assessor.md [hunk 19]
```diff
@@ -0,0 +1,66 @@
+# Customize-Assessor
+
+## Customize Assessor
+
+NNI also support building a assessor by yourself to adjust your tuning demand.
+
+If you want to implement a customized Assessor, there are three things for you to do:
+
+1) Inherit a assessor of a base Assessor class
```
Comment (xuehui1991) [inline docs/Customize_Assessor.md]: "a assessor" --> "an assessor" ??

File: docs/Builtin_Tuner.md [hunk 20]
```diff
@@ -0,0 +1,313 @@
+# Builtin Tuners
+
+NNI provides the-state-of-art tuning algorithm in our builtin-tuners and makes them easy to use. Below is the brief overview of NNI current builtin Tuners:
```
Comment (chicm-ms) [inline docs/Builtin_Tuner.md]: the-state-of-art => state-of-the-art ?

File: docs/Customize_Tuner.md [hunk 21]
```diff
@@ -0,0 +1,118 @@
+# Customize-Tuner
+
+## Customize Tuner
+
+NNI provides the-state-of-art tuning algorithm in our builtin-tuners. But we also support building a tuner by yourself to adjust your tuning demand.
```
Comment (chicm-ms) [inline docs/Customize_Tuner.md]: the-state-of-art => state-of-the-art ?

File: docs/Customize_Assessor.md [hunk 22]
```diff
@@ -0,0 +1,66 @@
...
+    ...
+    FLAGS, unparsed = parser.parse_known_args()
+
+    assessor = CustomizedAssessor(...)
+    assessor.run()
+
+main()
+```
+
+Please noted in **2**. The object `trial_history` are exact the object that Trial send to Assesor by using SDK `report_intermediate_result` function.
```
Comment (xuehui1991) [inline docs/Customize_Assessor.md]: "Assesor"  ---> "Assessor"

File: docs/Customize_Assessor.md [hunk 23]
```diff
@@ -0,0 +1,66 @@
...
+
+Also, user could override the `run` function in Assessor to control the process logic.
```
Comment (xuehui1991) [inline docs/Customize_Assessor.md]: "process logic" --> "processing logic"

File: docs/Customize_Tuner.md [hunk 24]
```diff
@@ -0,0 +1,118 @@
...
+
+If you want to implement and use your own tuning algorithm, you can implement a customized Tuner, there are three things for you to do:
+
+1) Inherit a tuner of a base Tuner class
+2) Implement receive_trial_result and generate_parameter function
+3) Configure your customized tuner in experiment yaml config file
```
Comment (xuehui1991) [inline docs/Customize_Tuner.md]: "yaml config"  --> "yml config" ??

File: docs/Trials.md [hunk 25]
```diff
@@ -0,0 +1,151 @@
+# Trials
+
+## Overview
+
+In NNI, hyper-parameter searching space will be firstly defined using a json file (often called searchspace.json). According to the search space defined by this file, different hyper-parameter combinations will be generated to try and get feedback information like final accurary.
```
Comment (xuehui1991) [inline docs/Trials.md]: "firstly defined " ---> “first defined ”？？

Comment (xuehui1991) [inline docs/Trials.md]: “accurary” --> "accuracy"

File: docs/Trials.md [hunk 26]
```diff
@@ -0,0 +1,151 @@
...
+
+A **Trial** in NNI is such an individual attempt at applying a set of hyper-parameters on a model. 
+
+
+
+NNI provide two approaches for you to define a trial: `NNI API` and `NNI Python annotation`.
```
Comment (xuehui1991) [inline docs/Trials.md]: "NNI provide two "  --> "NNI provides two "

File: docs/Trials.md [hunk 27]
```diff
@@ -0,0 +1,151 @@
...
+    useAnnotation: false
+    searchSpacePath: /path/to/your/search_space.json
+    ```
+
+- You can refer to [here](ExperimentConfig.md) for more information about how to set up experiment configurations.
+
+
+## NNI Python Annotation
+
+An alternative to write a trial is to use NNI's syntax for python. Simple as any annotation, NNI annotation is working like comments in your codes. You don't have to make structure or any other big changes to your existing codes. With a few lines of NNI annotation, you will be able to:
```
Comment (xuehui1991) [inline docs/Trials.md]: "write a trial"  --》 “writing a trial” ?

File: docs/Trials.md [hunk 28]
```diff
@@ -0,0 +1,151 @@
...
+Again, take MNIST as an example, it only requires 2 steps to write a trial with NNI Annotation.
+
+> Step 1 - Update codes with annotations 
+
+The following is a tensorflow code snippet for NNI Annotation, where the highlighted four lines are annotations that help you to: 
+  1. tune batch\_size and dropout\_rate
+  2. report test\_acc every 100 steps
+  3. at last report test\_acc as final result.
+
+What noteworthy is: as these new added codes are annotations, it does not actually change your previous codes logic, you can still run your code as usual in environments without NNI installed.
```
Comment (xuehui1991) [inline docs/Trials.md]: "these new added" --> "these newly added" ??

File: docs/Trials.md [hunk 29]
```diff
@@ -0,0 +1,151 @@
...
+
+**NOTE**: 
+- `@nni.variable` will take effect on its following line, which is an assignment statement whose leftvalue must be specified by the keyword `name` in `@nni.variable`.
+- `@nni.report_intermediate_result`/`@nni.report_final_result` will send the data to assessor/tuner at that line. 
+
+For more information about annotation syntax and its usage, please refer to [Annotation README](../tools/nni_annotation/README.md) . 
+
+
+>Step 2 - Enable NNI Annotation
+
```
Comment (xuehui1991) [inline docs/Trials.md]: "In the yaml configure file" -->"In the yml configure file" ??

File: docs/Trials.md [hunk 30]
```diff
@@ -0,0 +1,151 @@
...
+In the yaml configure file, you need to set *useAnnotation* to true to enable NNI annotation:
+```
+useAnnotation: true
+```
+
+## Others
+
+### Output
+
+NNI is designed to show each trial's stdout and stderr in the log region of Webui. However, currently this feature is supported only on PAI and k8s platforms. On other platforms NNI will show trial log path and users can temporarily check the output manually.
```
Comment (xuehui1991) [inline docs/Trials.md]: "region of Webui"  --> "region of WebUI"?

File: docs/gbdt_example.md [hunk 31]
```diff
@@ -0,0 +1,185 @@
+# GBDT in nni
+Gradient boosting is a machine learning technique for regression and classification problems, which produces a prediction model in the form of an ensemble of weak prediction models, typically decision trees. It builds the model in a stage-wise fashion as other boosting methods do, and it generalizes them by allowing optimization of an arbitrary differentiable loss function. 
+
+Gradient boosting decision tree has many popular implementation, such as [lightgbm](https://github.com/Microsoft/LightGBM), [xgboost](https://github.com/dmlc/xgboost), and [catboost](https://github.com/catboost/catboost), etc. GBDT is a great tool for solving the problem of traditional machine learning problem. Since GBDT is a robust algorithm, it could use in many domains. The better hyper-parameters for GBDT, the better performance you could achieve.
```
Comment (xuehui1991) [inline docs/gbdt_example.md]: "popular implementation" --> "popular implementations"??

File: docs/gbdt_example.md [hunk 32]
```diff
@@ -0,0 +1,185 @@
...
+
+Gradient boosting decision tree has many popular implementation, such as [lightgbm](https://github.com/Microsoft/LightGBM), [xgboost](https://github.com/dmlc/xgboost), and [catboost](https://github.com/catboost/catboost), etc. GBDT is a great tool for solving the problem of traditional machine learning problem. Since GBDT is a robust algorithm, it could use in many domains. The better hyper-parameters for GBDT, the better performance you could achieve.
+
+NNI is a great platform for tuning hyper-parameters, you could try various builtin search algorithm in nni and run multiple trials concurrently. 
+
+
+## 1. Search Space in GBDT
+There are many hyper-parameters in GBDT, but what kind of parameters will effect the performance or speed? Based on some pratical experience, some suggestion here(Take lightgbm as example):
```
Comment (xuehui1991) [inline docs/gbdt_example.md]: " effect the performance" --> "affect the performance" ??

Comment (xuehui1991) [inline docs/gbdt_example.md]: "pratical experience" --> "practical experience"?

File: docs/gbdt_example.md [hunk 33]
```diff
@@ -0,0 +1,185 @@
+# GBDT in nni
+Gradient boosting is a machine learning technique for regression and classification problems, which produces a prediction model in the form of an ensemble of weak prediction models, typically decision trees. It builds the model in a stage-wise fashion as other boosting methods do, and it generalizes them by allowing optimization of an arbitrary differentiable loss function. 
+
+Gradient boosting decision tree has many popular implementation, such as [lightgbm](https://github.com/Microsoft/LightGBM), [xgboost](https://github.com/dmlc/xgboost), and [catboost](https://github.com/catboost/catboost), etc. GBDT is a great tool for solving the problem of traditional machine learning problem. Since GBDT is a robust algorithm, it could use in many domains. The better hyper-parameters for GBDT, the better performance you could achieve.
+
+NNI is a great platform for tuning hyper-parameters, you could try various builtin search algorithm in nni and run multiple trials concurrently.
```
Comment (xuehui1991) [inline docs/gbdt_example.md]: "ake lightgbm as example" --> "ake lightgbm as an example"

File: docs/gbdt_example.md [hunk 34]
```diff
@@ -0,0 +1,185 @@
...
+NNI is a great platform for tuning hyper-parameters, you could try various builtin search algorithm in nni and run multiple trials concurrently. 
+
+
+## 1. Search Space in GBDT
+There are many hyper-parameters in GBDT, but what kind of parameters will effect the performance or speed? Based on some pratical experience, some suggestion here(Take lightgbm as example):
+
+> * For better accuracy
+* `learning_rate`. The range of `learning rate` could be [0.001, 0.9].
+
+* `num_leaves`. `num_leaves` is realted to `max_depth`, you don't have to tune both of them.
```
Comment (xuehui1991) [inline docs/gbdt_example.md]: "realted to max_depth" --> "related to max_depth"

File: docs/gbdt_example.md [hunk 35]
```diff
@@ -0,0 +1,185 @@
...
+* `num_iterations`. May larger if underfitting.
+
+> * For speed up
+* `bagging_fraction`. The range of `bagging_fraction` could be [0.7, 1.0].
+
+* `feature_fraction`. The range of `feature_fraction` could be [0.6, 1.0].
+    
+* `max_bin`.
+
+> * For avoid overfitting
```
Comment (xuehui1991) [inline docs/gbdt_example.md]: "For avoid overfitting" --> "To avoid overfitting"??

File: docs/gbdt_example.md [hunk 36]
```diff
@@ -0,0 +1,185 @@
...
+* `min_data_in_leaf`. This depend on your dataset.
+
+* `min_sum_hessian_in_leaf`. This depend on your dataset.
```
Comment (xuehui1991) [inline docs/gbdt_example.md]: "This depend on your dataset" --> " This depends on your dataset"

File: docs/gbdt_example.md [hunk 37]
```diff
@@ -0,0 +1,185 @@
...
+
+* `num_leaves`.
+
+Reference link:
+[lightgbm](https://lightgbm.readthedocs.io/en/latest/Parameters-Tuning.html) and 
+[autoxgoboost](https://github.com/ja-thomas/autoxgboost/blob/master/poster_2018.pdf)
+
+## 2. Task description
+Now we come back to our example "auto-gbdt" which run in lightgbm and nni. The data including [train data](https://github.com/Microsoft/nni/blob/master/examples/trials/auto-gbdt/data/regression.train) and [test data](https://github.com/Microsoft/nni/blob/master/examples/trials/auto-gbdt/data/regression.train). 
+Given the features and label in train data, we train a GBDT regession model and use it to predict.
```
Comment (xuehui1991) [inline docs/gbdt_example.md]: " regession"-> " regression"

File: docs/SQuAD_evolution_examples.md [hunk 38]
```diff
@@ -0,0 +1,308 @@
...
+This example shows us how to use Genetic Algorithm to find good model architectures for Reading Comprehension.
+
+## 1. Search Space
+Since attention and recurrent neural network (RNN) have been proven effective in Reading Comprehension.
+We conclude the search space as follow:
+
+1. IDENTITY (Effectively means keep training).
+2. INSERT-RNN-LAYER (Inserts a LSTM. Comparing the performance of GRU and LSTM in our experiment, we decided to use LSTM here.)
+3. REMOVE-RNN-LAYER
+4. INSERT-ATTENTION-LAYER(Inserts a attention layer.)
```
Comment (xuehui1991) [inline docs/SQuAD_evolution_examples.md]: "a attention layer" --> "an attention layer"

File: docs/SQuAD_evolution_examples.md [hunk 39]
```diff
@@ -0,0 +1,308 @@
...
+
+`trialConcurrency` is the number of trials running concurrently, which is the number of GPUs you want to use, if you are setting `gpuNum` to 1.
+
+### 3.2 submit this job
+
+```
+nnictl create --config ~/nni/examples/trials/ga_squad/config_pai.yml
+```
+
+## 4. Techinal details about the trial
```
Comment (xuehui1991) [inline docs/SQuAD_evolution_examples.md]: "Techinal details" --> "Technical details"

File: docs/SQuAD_evolution_examples.md [hunk 40]
```diff
@@ -0,0 +1,308 @@
...
+The trial has a lot of different files, functions and classes. Here we will only give most of those files a brief introduction:
+
+* `attention.py` contains an implementation for attention mechanism in Tensorflow.
+* `data.py` contains functions for data preprocessing.
+* `evaluate.py` contains the evaluation script.
+* `graph.py` contains the definition of the computation graph.
+* `rnn.py` contains an implementation for GRU in Tensorflow.
+* `train_model.py` is a wrapper for the whole question answering model.
+
+Among those files, `trial.py` and `graph_to_tf.py` is special.
```
Comment (xuehui1991) [inline docs/SQuAD_evolution_examples.md]: " is special" --> "are special"??

File: docs/SQuAD_evolution_examples.md [hunk 41]
```diff
@@ -0,0 +1,308 @@
...
+            "size": "y",
+            "output": [],
+            "is_delete": false
+        },
+        {"Comment": "More layers will be here for actual graphs."}
+    ]
+}
+```
+
+Every model configuration will has a "layers" section, which is a JSON list of layer definitions. The definition of each layer is also a JSON object, where:
```
Comment (xuehui1991) [inline docs/SQuAD_evolution_examples.md]: "has a "layers" section" --> "have a "layers" section" ??

File: docs/SQuAD_evolution_examples.md [hunk 42]
```diff
@@ -0,0 +1,308 @@
...
+
+ * `type` is the type of the layer. 0, 1, 2, 3, 4 correspond to attention, self-attention, RNN, input and output layer respectively.
+ * `size` is the length of the output. "x", "y" correspond to document length / question length, respectively.
```
Comment (xuehui1991) [inline docs/SQuAD_evolution_examples.md]: "correspond to" --> "corresponds to"??

File: docs/Customize_Tuner.md (hunk 21 above)
Comment (chicm-ms) [inline docs/Customize_Tuner.md]: Remove "but" or use "and" instead?

File: docs/Customize_Tuner.md (hunk 24 above)
Comment (leckie-chn) [reply docs/Customize_Tuner.md]: yaml is also legal

File: docs/Trials.md (hunk 29 above)
Comment (leckie-chn) [reply docs/Trials.md]: see above

File: docs/gbdt_example.md (hunk 33 above)
Comment (leckie-chn) [reply docs/gbdt_example.md]: ???

File: docs/Builtin_Tuner.md (hunk 3 above)
Comment (chicm-ms) [reply docs/Builtin_Tuner.md]: @QuanluZhang

**Pull request merged**